File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/bucinfo/BucInfo.java
package no.nav.melosys.eessi.models.bucinfo;

import lombok.Data;

@Data
public class BucInfo {

    private static final String PROCESS_OWNER = "PO";

    private String applicationRoleId;
    private String id;
    private String processDefinitionId;
    private String status;

    public boolean norgeErCaseOwner() {
        return PROCESS_OWNER.equalsIgnoreCase(applicationRoleId);
    }

    public boolean bucErÅpen() {
        return "open".equalsIgnoreCase(status);
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/bucinfo/BucInfo.kt
package no.nav.melosys.eessi.models.bucinfo

data class BucInfo(
    var applicationRoleId: String? = null,
    var id: String? = null,
    var processDefinitionId: String? = null,
    var status: String? = null
) {
    fun norgeErCaseOwner(): Boolean = PROCESS_OWNER.equals(applicationRoleId, ignoreCase = true)

    // TODO: put "open" i en const eller bruke en hjelpe classe for disse statusene
    fun bucErÅpen(): Boolean = "open".equals(status, ignoreCase = true)

    companion object {
        private const val PROCESS_OWNER = "PO"
    }
}
